apps/managedidentity/managedidentity.go

```diff
@@ -76,59 +79,154 @@ func (s systemAssignedValue) value() string { return string(s) }
 	for _, option := range options {
-		option.apply(&opts)
+		option(&opts)
 	}
 
-	authInfo, err := authority.NewInfoFromAuthorityURI("authorityURI", true, false)
+	client := Client{ // TODO :: check for http client
+		MiType:     id,
+		httpClient: opts.httpClient,
+	}
+
+	return client, nil
+}
+
+func getTokenForURL(url *url.URL, httpClient ops.HTTPClient) (accesstokens.TokenResponse, error) {
+	req, err := http.NewRequest(http.MethodGet, url.String(), nil)
 	if err != nil {
-		return Client{}, err
+		return accesstokens.TokenResponse{}, err
 	}
+	req.Header.Add("Metadata", "true")
 
-	authParams := authority.NewAuthParams(id.value(), authInfo)
-	client := Client{ // Note: Hey, don't even THINK about making Base into *Base. See "design notes" in public.go and confidential.go
-		AuthParams:      authParams,
-		cacheAccessorMu: &sync.RWMutex{},
-		// manager:         storage.New(token),
-		// pmanager:        storage.NewPartitionedManager(token),
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return accesstokens.TokenResponse{}, err
 	}
 
-	return client, err
+	// Pull out response body
+	responseBytes, err := io.ReadAll(resp.Body)
+	defer resp.Body.Close()
+	if err != nil {
+		return accesstokens.TokenResponse{}, err
+	}
+
+	// Unmarshall response body into struct
+	var r accesstokens.TokenResponse
+	err = json.Unmarshal(responseBytes, &r)
+	if err != nil {
+		return accesstokens.TokenResponse{}, err
+	}
+	return r, nil
+
 }
 
 // Acquires tokens from the configured managed identity on an azure resource.
 //
 // Resource: scopes application is requesting access to
 // Options: [WithClaims]
 func (client Client) AcquireToken(context context.Context, resource string, options ...AcquireTokenOption) (base.AuthResult, error) {
-	return base.AuthResult{}, nil
+	o := AcquireTokenOptions{}
+
+	for _, option := range options {
+		option(&o)
+	}
+
+	// try and find some resource which can be accessed
+	// service fabric  GET
+	// app service  GET
+	// could shell  POST request
+	// azure arc  GET
+	// default :: IMDS  GET
+	//
+	// Sources that send GET requests: App Service, Azure Arc, IMDS, Service Fabric
+	//
+	// Sources that send POST requests: Cloud Shell
+
+	var msiEndpoint *url.URL
+	msiEndpoint, err := url.Parse(IMDSEndpoint)
+	if err != nil {
+		fmt.Println("Error creating URL: ", err)
+		return base.AuthResult{}, nil
+	}
+	msiParameters := msiEndpoint.Query()
+	msiParameters.Add("api-version", "2018-02-01")
+	msiParameters.Add("resource", resource)
+
+	if len(o.Claims) > 0 {
+		msiParameters.Add("claims", o.Claims)
+	}
+
+	switch client.MiType.(type) {
+	case ClientID:
+		msiParameters.Add(MIQuerryParameterClientId, client.MiType.value())
+	case ResourceID:
+		msiParameters.Add(MIQuerryParameterResourceId, client.MiType.value())
+	case ObjectID:
+		msiParameters.Add(MIQuerryParameterObjectId, client.MiType.value())
+	case systemAssignedValue: // not adding anything
+	default:
+		return base.AuthResult{}, fmt.Errorf("Type not suported")
```

Comment:
Would be helpful to note the type name
```suggestion
		return base.AuthResult{}, fmt.Errorf("unsupported type %T", client.miType)
```